scripts/generate_psa_key_attributes.py

```diff
@@ -0,0 +1,238 @@
+from pathlib import Path
+
+class PsaKeyType(IntEnum):
+    """The type of the key"""
+
+    AES = 0x2400
+    ECC_TWISTED_EDWARDS = 0x4142
+    RAW_DATA = 0x1001
+
+
+class PsaKeyBits(IntEnum):
+    """Number of bits in the key"""
+
+    AES = 256
+    EDDSA = 255
+
+
+class PsaUsage(IntEnum):
+    """Permitted usage on a key"""
+
+    VERIFY_MESSAGE_EXPORT = 0x0801
+    ENCRYPT_DECRYPT = 0x0300
+    USAGE_DERIVE = 0x4000
+
+class PsaCracenUsageSceme(IntEnum):
+    NONE = 0xff
+    PROTECTED = 0
+    SEED = 1
+    ENCRYPTED = 2
+    RAW = 3
+
+class PsaKeyLifetime(IntEnum):
+    """Lifetime and location for storing key"""
+
+    PERSISTENT_CRACEN = 0x804E0001
+    PERSISTENT_CRACEN_KMU = 0x804E4B01
+
+
+class PsaAlgorithm(IntEnum):
+    """Algorithm that can be associated with a key. Not used for AES"""
+
+    NONE = 0
+    CBC = 0x04404000
+    EDDSA_PURE = 0x06000800
+
+
+class PlatformKeyAttributes:
+    def __init__(self,
+                 key_type: PsaKeyType,
+                 identifier: int,
+                 location: PsaKeyLifetime,
+                 usage: PsaUsage,
+                 algorithm: PsaAlgorithm,
+                 size: int,
+                 cracen_usage: PsaCracenUsageSceme = PsaCracenUsageSceme.NONE):
+
+        self.key_type = key_type
+        self.lifetime = location
+        self.usage = usage
+        self.alg0 = algorithm
+
+        if location == PsaKeyLifetime.PERSISTENT_CRACEN_KMU:
+            if cracen_usage == PsaCracenUsageSceme.NONE:
+                print("--cracen_usage must be set if location target is PERSISTENT_CRACEN_KMU")
+                return
+            self.identifier = 0x7fff0000 | (cracen_usage << 12) | (identifier & 0xff)
+        else:
+            self.identifier = identifier
+
+        self.bits = size
+
+        if self.key_type == PsaKeyType.AES:
+            self.alg1 = PsaAlgorithm.NONE
+        elif self.key_type == PsaKeyType.ECC_TWISTED_EDWARDS:
+            self.alg1 = PsaAlgorithm.NONE
+        elif self.key_type == PsaKeyType.RAW_DATA:
+            self.alg1 = PsaAlgorithm.NONE
+        else:
+            raise RuntimeError("Invalid key type")
+
+    def pack(self):
+        """Builds a binary blob compatible with the psa_key_attributes_s C struct"""
+
+        return struct.pack(
+            "<hhIIIIII",
+            self.key_type,
+            self.bits,
+            self.lifetime,
+            # Policy
+            self.usage,
+            self.alg0,
+            self.alg1,
+            self.identifier,
+            0,  # Reserved, only used if key id encodes owner id
+        )
+
+def is_valid_hexa_code(string):
+    try:
+        int(string, 16)
+        return True
+    except ValueError:
+        return False
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Generate PSA key attributes and write to stdout."
+        "Use 'xxd' or similar if you want to print it to a terminal:"
+        f" 'python {Path(__file__).name} --args | xxd'",
```

Comment:
This needs to be updated to match the new functionality and intended use.

Reply:
ok